pkg/bot.go

```diff
@@ -113,9 +118,36 @@ type kubeClients struct {
 	// core API client and lister
 	coreClientSet *kubernetes.Clientset
 	podLister     corelister.PodLister
+
+	// gRPC auth admin service
+	authClient pbAdmin.AdministrationClient
+}
+
+func authAdminAPI(k *kubernetes.Clientset, authSVCLabels map[string]string) (pbAdmin.AdministrationClient, error) {
+	svcs, err := k.CoreV1().Services("").List(metav1.ListOptions{LabelSelector: labels.FormatLabels(authSVCLabels)})
+	if err != nil {
+		return nil, err
+	}
+	if len(svcs.Items) == 0 {
+		return nil, fmt.Errorf("failed to lookup admin gRPC service for labels %v, expected exactly 1, found %v", authSVCLabels, len(svcs.Items))
```

Comment:
what do you have in mind? currently, it will error out and the operator using this codebase exit with error message. Open to suggestions but this makes the most sense to me. If there are multiple services claiming to be `admin-api`, then we have likely broken deployment and need to take immediate action, this doesn't happen by accident.